### ISSUE TITLE
Document precisely what "detect_accelerometers" does

### ISSUE THREAD

**Issue by mystica555** [alesguzik/linux_detect_tablet_mode#1]
Hello,

I have a convertible Toshiba Satellite Radius P25W-C2300-4k (yes a 12.5 4k screen; hidpi is fun) 

I want to know what the accelerometers are doing but I also want to know if I actually _have any_ to begin with that expose themselves to Linux.  

I found your script; the "detect_accelerometers" seems to do "something".

It outputs "4"

What precisely does "4" mean?

**Comment by alesguzik** [comment]
Hi.

TL;DR: most likely your device have only 1 accelerometer, so this scripts won't work for you.

On Linux accelerometers are handled by iio subsystem and so are exposed as directories like `/sys/bus/iio/devices/iio:device0`, `/sys/bus/iio/devices/iio:device1`, ... alongside other iio devices (accelerometers have a file `/sys/bus/iio/devices/iio:device*/name` with content `accel_3d`). Unfortunately, initialization order of iio devices is not fixed, so if your keyboard accelerometer is available as `/sys/bus/iio/devices/iio:device1` right now, it may become `/sys/bus/iio/devices/iio:device4` after reboot. This means that when `watch_tablet` starts it needs to detect accelerometers.

This set of scripts will work only if your laptop has 2 accelerometers - one in a base and another one in a lid. `cat /sys/bus/iio/devices/iio:device*/name|grep accel_3d|wc -l` returns the number of accelerometers, and it has to return 2. If it will return 0 or 1 this set of scripts is useless, as it works by calculating an angle between 2 acceleration vectors. If it will return 3 or more, then `detect_accelerometers` script have to be changed to find correct 2 accelerometers.

Right now it just assumes that there will be exactly 2 accelerometer iio devices found and assumes that `detect_accelerometers` is ran by `watch_tablet` script during initialization (which is assumed to be ran when you power on your laptop and start desktop session, so your laptop is most likely in normal, non-tablet mode). It then grabs first 2 accelerometers in whichever order they apper, runs `detect_tablet_mode` script once with this 2 accelerometer device ids, and if it returns 'tablet' instead of 'normal' then it swaps the order of accelerometers before returning them, so when `detect_tablet_mode` is later called by `watch_tablet`, initial mode detection would result in `normal`.

**Comment by alesguzik** [comment]
@mystica555 I reread your question and it seems I answered not what you have asked about. detect_accelerometers returns space-separated list of accelerometer iio device ids (possibly reordered for the watch_tablet script), so `4` means you have a single accelerometer at `/sys/bus/iio/devices/iio:device4/`, which means your device won't work with this set of scripts.
